### ISSUE TITLE
Sell Loot Metals/Stones Matches Deeds

### ISSUE THREAD

**Issue by wstampley** [rpherbig/dr-scripts#4972]
```
You rummage about your person, looking for something...
 
  Your oravir deed is in an oiled leather traveler's pack.
  Your covellite deed is in an oiled leather traveler's pack.
  Your covellite deed is in an oiled leather traveler's pack.
  Your covellite deed is in an oiled leather traveler's pack.
  Your covellite deed is in an oiled leather traveler's pack.
  Your covellite deed is in an oiled leather traveler's pack.
 
 ....
 
 [sell-loot]>get my oravir nugget from my pack
What were you referring to?
>
[sell-loot]>sell my oravir nugget to Veilex
Sell what?
>
[sell-loot]>get my covellite nugget from my pack
What were you referring to?
>
[sell-loot]>sell my covellite nugget to Veilex
>
[sell-loot]>get my covellite nugget from my pack
Sell what?
>
[sell-loot]>sell my covellite nugget to Veilex
What were you referring to?
>
[sell-loot]>get my covellite nugget from my pack
Sell what?
>
[sell-loot]>sell my covellite nugget to Veilex
What were you referring to?
>
[sell-loot]>get my covellite nugget from my pack
Sell what?
>
[sell-loot]>sell my covellite nugget to Veilex
What were you referring to?
>
[sell-loot]>get my covellite nugget from my pack
Sell what?
>
[sell-loot]>sell my covellite nugget to Veilex
What were you referring to?
```

**Comment by KatoakDR** [comment]
Thanks for reporting this

**Comment by KatoakDR** [comment]
I haven't forgotten about this, it's on my todo list

**Comment by KatoakDR** [comment]
@wstampley I created https://regex101.com/r/5lFgKc/1 to test out the regex matching to the rummage info you provided but it's not matching the deeds.

Is there any other info you can provide for how I could test/reproduce this?

Thanks

**Comment by KatoakDR** [comment]
Well, I realize now that the rummage info you shared was you searching and showing your deeds. Was the original rummage when the sell-loot script run something like this:

```
You rummage through an oiled leather traveler's pack and see a deed for oravir nuggets, a deed for covellite nuggets and a deed for covellite nuggets.
```

Because if so, then I do see how that'd match.

**Comment by KatoakDR** [comment]
Updating the regex to explicitly specify the nugget/bar sizes seems to resolve the issue: https://regex101.com/r/5lFgKc/3

**Comment by KatoakDR** [comment]
Here's where I'm at with refactoring the `sell_metals_and_stones` method in `sell-loot`:

```ruby
  def sell_metals_and_stones(container)
    DRC.release_invisibility

    item_data = get_data('items')
    metal_types = item_data['metal_types']
    stone_types = item_data['stone_types']
    ignore_types = @settings.sell_loot_ignored_metals_and_stones

    metals_regex = /\b(?<size>tiny|small|medium|large|huge|massive|enormous) (?<material>#{metal_types.join('|')}) (?<noun>nugget|bar)\b/i
    stones_regex = /\b(?<material>#{stone_types.join('|')}) (?<noun>pebble|stone|rock|boulder)\b/i
    deeds_regex = /deed for/i
    ignored_regex = /\b#{ignore_types.join('|')}\b/i unless ignore_types.empty?

    items = DRCI.get_item_list(container)
      .reject { |item| item =~ deeds_regex }
      .reject { |item| ignored_regex ? item =~ ignored_regex : false }
      .reduce([]) do |items_to_sell, item|
        if item =~ metals_regex || item =~ stones_regex
          material = Regexp.last_match[:material]
          noun = Regexp.last_match[:noun]
          items_to_sell << "#{material} #{noun}"
        end
        items_to_sell
      end

    unless items.empty?
      return unless DRCT.walk_to(@hometown['gemshop']['id'])
      clerk = which_clerk(@hometown['gemshop']['name'])

      items.each do |item|
        fput("get my #{item} from my #{container}")
        fput("sell my #{item} to #{clerk}")
      end
    end
  end
```

**Comment by wstampley** [comment]
Not sure if this is helpful:

```
>;e echo DRCI.get_item_list('pack')
--- Lich: exec1 active.
[exec1]>rummage my pack
You rummage through an oiled leather traveler's pack and see a deed for a large covellite nugget, a packet of deed claim forms, a deed for a small covellite nugget, a deed for a huge covellite nugget, a pair of white ice skates with fine brass buckles, a steel greataxe, a simple leather sling, a steel throwing mallet, a light steel throwing axe, a diamondwood nightstick inlaid with cobalt, a simply carved maple stonebow with a bright blue leather sling, a steel akabo, a spiked steel hammer, a steel battle axe, a glaes light spear, a silverwood Nisha shortbow, a paper-wrapped textbook, a jug of oil, a deed register, a large packet of deed claim forms, a deed for a covellite ingot, a deed for a lumium ingot, a deed for a steel ingot, a book of master weaponsmithing instructions, an iron wire brush, a small covellite nugget, a massive covellite nugget, a large oravir nugget, a kirmhiro draught, a sigil-stamped goblet filled with wine-hued carnelian, some borax flux, a tiny clear diamond, an acanth bank book with openwork carvings, a soft chamois cloth with golden trim, a cleaning cloth, a furtive summons, a set of animite toe-bells inlaid with blue glass raindrops, a steel carving knife, a flat steel shaper, a square steel shovel, a deep viridian verdant heart, some borax flux, a worn book bound in lemon-yellow dimity, a worn book bound in teal taffeta, a deed for a massive oravir nugget, a small covellite nugget, a sleek goldenoak writing quill tipped with a long royal-blue plume, a book of master armorsmithing instructions, a steel akabo, a massive covellite nugget, a silverwood vault book with openwork carvings, a red-scaled wyvern with firestained indurium and blood ruby eyes, some bundling rope, a pair of heavily reinforced trousers fastened by glass eyeballs, a glaes rapier, a book of master blacksmithing instructions, a flask of oil, a forging work order logbook, a basic stamp and some muracite plain pliers.
>
[exec1: ["deed for ", "large covellite nugget", "packet of deed claim forms", "deed for ", "small covellite nugget", "deed for ", "huge covellite nugget", "pair of white ice skates with fine brass buckles", "steel greataxe", "simple leather sling", "steel throwing mallet", "light steel throwing axe", "diamondwood nightstick inlaid with cobalt", "simply carved maple stonebow with ", "bright blue leather sling", "steel akabo", "spiked steel hammer", "steel battle axe", "glaes light spear", "silverwood Nish", "shortbow", "paper-wrapped textbook", "jug of oil", "deed register", "large packet of deed claim forms", "deed for ", "covellite ingot", "deed for ", "lumium ingot", "deed for ", "steel ingot", "book of master weaponsmithing instructions", "iron wire brush", "small covellite nugget", "massive covellite nugget", "large oravir nugget", "kirmhiro draught", "sigil-stamped goblet filled with wine-hued carnelian", "borax flux", "tiny clear diamond", "acanth bank book with openwork carvings", "soft chamois cloth with golden trim", "cleaning cloth", "furtive summons", "set of animite toe-bells inlaid with blue glass raindrops", "steel carving knife", "flat steel shaper", "square steel shovel", "deep viridi", "verdant heart", "borax flux", "worn book bound in lemon-yellow dimity", "worn book bound in teal taffeta", "deed for ", "massive oravir nugget", "small covellite nugget", "sleek goldenoak writing quill tipped with ", "long royal-blue plume", "book of master armorsmithing instructions", "steel akabo", "massive covellite nugget", "silverwood vault book with openwork carvings", "red-scaled wyvern with firestained indurium and blood ruby eyes", "bundling rope", "pair of heavily reinforced trousers fastened by glass eyeballs", "glaes rapier", "book of master blacksmithing instructions", "flask of oil", "forging work order logbook", "basic stamp", "muracite plain pliers"]]
```

**Comment by KatoakDR** [comment]
Yes, thank you, that's very helpful and definitely pin pointed where the issue was :)

**Comment by asechrest** [comment]
I've confirmed this still happens. I'll work on fixing this one.

**Comment by asechrest** [comment]
@rpherbig Can close via my merged PR #5339
